criteria.c

```diff
@@ -82,7 +83,7 @@ bool match_criteria(struct mako_criteria *criteria,
 	}
 
 	if (spec.summary &&
-			strcmp(criteria->summary, notif->summary) != 0) {
+			regexec(&criteria->summary, notif->summary, 0, NULL, 0) == REG_NOMATCH) {
```

Comment:
I think your option of spitting an error out potential errors on stderr is fine at least for the first pass. We don't really have a more robust error reporting mechanism in place, and most users will be keeping an eye on stderr while creating new configuration anyway.